sycl/include/sycl/ext/oneapi/experimental/cuda/barrier.hpp

```diff
@@ -33,6 +33,7 @@ class barrier {
   barrier(barrier &&other) noexcept = delete;
   barrier &operator=(const barrier &other) = delete;
   barrier &operator=(barrier &&other) noexcept = delete;
+  ~barrier() = default;
```

Comment:
this class only has a state member variable, so the default destructor should be fine.